psi4/driver/qcdb/vib.py

```diff
@@ -0,0 +1,1062 @@
+                        text += """{:{colsp}}""".format('', colsp=colsp)
+                    text += '\n'
+
+    return text
+
+
+def thermo(vibinfo, T, P, multiplicity, molecular_mass, E0, sigma, rot_const, rotor_type=None):
+    """Perform thermochemical analysis from vibrational output.
+
+    Parameters
+    ----------
+    E0 : float
+        Electronic energy [Eh] at well bottom at 0 [K], :psivar:`CURRENT ENERGY`.
+    molecular_mass : float
+        Mass in [u] of molecule under analysis.
+    multiplicity: int
+        Spin multiplicity of molecule under analysis.
+    rot_const : ndarray of floats
+        (3,) rotational constants in [cm^-1] of molecule under analysis.
+    sigma : int
+        The rotational or external symmetry number determined from the point group.
+    rotor_type : str
+        The rotor type for rotational stat mech purposes: RT_ATOM, RT_LINEAR, other.
+    T : float
+        Temperature in [K]. Psi default 298.15. Note that 273.15 is IUPAC STP.
+    P : float
+        Pressure in [Pa]. Psi default 101325. Note that 100000 is IUPAC STP.
+
+    Returns
+    -------
+    dict of VibrationAspect, str
+        First is every thermochemistry component in atomic units.
+        Second is formatted presentation of analysis.
+
+    """
+    import math
+    import collections
+
+    sm = collections.defaultdict(float)
+
+    # electronic
+    q_elec = multiplicity
+    sm[('S', 'elec')] = math.log(q_elec)
+
+    # translational
+    beta = 1 / (psi_kb * T)
+    q_trans = (2.0 * np.pi * molecular_mass * psi_amu2kg / (beta * psi_h * psi_h))**1.5 * psi_na / (beta * P)
+    sm[('S', 'trans')] = 5 / 2 + math.log(q_trans / psi_na)
+    sm[('Cv', 'trans')] = 3 / 2
+    sm[('Cp', 'trans')] = 5 / 2
+    sm[('E', 'trans')] = 3 / 2 * T
+    sm[('H', 'trans')] = 5 / 2 * T
+
+    # rotational
+    if rotor_type == "RT_ATOM":
+        pass
+    elif rotor_type == "RT_LINEAR":
+        q_rot = 1. / (beta * sigma * 100 * psi_c * psi_h * rot_const[1])
+        sm[('S', 'rot')] = 1.0 + math.log(q_rot)
+        sm[('Cv', 'rot')] = 1
+        sm[('Cp', 'rot')] = 1
+        sm[('E', 'rot')] = T
+    else:
+        phi_A, phi_B, phi_C = rot_const * 100 * psi_c * psi_h / psi_kb
+        q_rot = math.sqrt(math.pi) * T**1.5 / (sigma * math.sqrt(phi_A * phi_B * phi_C))
+        sm[('S', 'rot')] = 3 / 2 + math.log(q_rot)
+        sm[('Cv', 'rot')] = 3 / 2
+        sm[('Cp', 'rot')] = 3 / 2
+        sm[('E', 'rot')] = 3 / 2 * T
+    sm[('H', 'rot')] = sm[('E', 'rot')]
+
+    # vibrational
+    vibonly = filter_nonvib(vibinfo)
+    ZPE_cm_1 = 1 / 2 * np.sum(vibonly['omega'].data.real)
+    omega_str = _format_omega(vibonly['omega'].data, decimals=4)
+
+    imagfreqidx = np.where(vibonly['omega'].data.imag > vibonly['omega'].data.real)[0]
+    if len(imagfreqidx):
+        print("Warning: thermodynamics relations excluded imaginary frequencies: {}".format(omega_str[imagfreqidx]))
```

Comment:
If there's exactly 1 imaginary frequency, perhaps we could think about automagically reporting the [Wigner tunneling correction](https://reactionmechanismgenerator.github.io/RMG-Py/reference/kinetics/wigner.html).

Reply:
@CDSherrill already has plans of putting someone on the Grimme low-freq rotor correction. Wigner's another good one.

Reply:
I am assuming this (Wigner tunnelling or hindered rotor treatments) more or less fell through the cracks? Or do you have anyone working on it?

Reply:
It did make it on to the wish list, so not wholly lost. No one working on it. @rainli323 wanted a project and it might be suitably compact unless you had your eye on it.